### PR TITLE
Enable point-in-time recovery for dynamodb table.

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
When using `tfsec` to analyze this code it comes back with this warning:

```

      42 | # Terraform state locking
      43 | #
      44 | 
      45 | resource "aws_dynamodb_table" "terraform_state_lock" {
      46 |   name           = var.dynamodb_table_name
      47 |   hash_key       = "LockID"
      48 |   read_capacity  = 2
      49 |   write_capacity = 2
      50 | 
      51 |   server_side_encryption {
      52 |     enabled = true
      53 |   }
      54 | 
      55 |   attribute {
      56 |     name = "LockID"
      57 |     type = "S"
      58 |   }
      59 | 
      60 |   tags = var.dynamodb_table_tags
      61 | }
      62 | 

  Impact:     Accidental or malicious writes and deletes can't be rolled back
  Resolution: Enable point in time recovery

  https://tfsec.dev/docs/aws/AWS086/ 
  https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dynamodb_table#point_in_time_recovery 
  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/PointInTimeRecovery.html 
```

The fix is simple and feels like a good default for folks.

### CODE REVIEW

**Comment by chrisgilmerproj** [comment]
@dynamike @eeeady - could I get a review on this one?

**Comment by dynamike** [comment]
Let me chat around with folks on this. My initial reaction is we don't really care about the data as they are only ephemeral locks and can't really picture a need to do a real recovery.

**Comment by chrisgilmerproj** [comment]
I agree, this doesn't really move the needle. The `tfsec` tool is fickle and its hard to ignore problems in modules as dependencies. Alternatively we can make this a var and let folks choose if they care. I'm not sure the cost is a big issue here given that the scale of usage.

**Comment by dynamike** [comment]
A conditional variable is one option. I would also be okay with adding a tfsec specific annotation with a comment describing why the finding is not applicable for this use case.

**Comment by chrisgilmerproj** [comment]
@dynamike - I've taken both of your suggestions here. There is now a variable to let folks choose for themselves but it defaults to `false` which was the old behavior. I've also added an explicit ignore for the tfsec rule.

I'm not sure why I can't get the checks to pass but it might be a ghactions issue on my repo's end for the fork. Let me know if I need to do something about it.

File: main.tf
Comment (dynamike) [inline main.tf:45]: Can you add a bit more context on why we would be disabling the AWS086 finding?

File: variables.tf
Comment (dynamike) [inline variables.tf:62]: To make it clear we are enabling point in time recovery for dynamodb
```suggestion 
variable "dynamodb_point_in_time_recovery" {
```

File: main.tf
Comment (dynamike) [inline main.tf:62]:
```suggestion
    enabled = var.dynamodb_point_in_time_recovery
```

**Review by dynamike — CHANGES_REQUESTED** [review]

**Comment by dynamike** [comment]
For the GH actions permission error, there should be a setting to enable the action in your forked PR. I don't have a specific example to show you, but let me know if you can't find it.

**Comment by chrisgilmerproj** [comment]
> For the GH actions permission error, there should be a setting to enable the action in your forked PR. I don't have a specific example to show you, but let me know if you can't find it.

I went ahead and ensured that actions are enabled on my fork of the repo. But it appears that's not enough. I did some searching and found this thread that indicates maybe a change is needed in the workflow? I could be wrong but its worth maybe checking out:

https://github.com/stefanzweifel/git-auto-commit-action/issues/114 

Let me know what you need me to do to get this rolling.

**Comment by dynamike-debug** [comment]
Can you try merging changes from the `main` branch. It looks likes the terraform-docs action wasn't working, but was fixed after an update.

**Comment by dynamike** [comment]
Can you try merging in changes from `main` into your fork? I think there was an issue with the terraform-docs github action, but it appears to be working now.

**Comment by chrisgilmerproj** [comment]
@dynamike - I'm surprised this isn't working even after merging in `master`. I'm fairly certain I did the right thing too. Any thoughts on what's going on here? Are there odd permissions against the remote repo I can look into?

**Comment by dynamike** [comment]
Yeah there are some rough edges to workout. The first place to check is the actions tab in your forked repo. There should be a button to enable actions on the forked repo. Something like `Workflows aren't being run on this forked repository`. 

If that doesn't work then I'd next look at a more dubious change around allowing edits by maintainers in the PR you submitted. 
image

**Comment by chrisgilmerproj** [comment]
Here is my actions tab in the forked repo:

Screen Shot 2021-08-03 at 2 04 51 PM

**Comment by dynamike** [comment]
Multiple actions pages makes this a fun exercise. It should be under https://github.com/dod-iac/terraform-aws-bootstrap/actions

**Comment by chrisgilmerproj** [comment]
Aha, I see how that was confusing. So I went to the tab and the workflow is enabled, it just never has run in my fork.

Screen Shot 2021-08-03 at 2 16 48 PM

**Comment by dynamike** [comment]
Thanks for debugging with me. That all looks correct. I think the next thing to look at is to allow edits and access to maintainers, because the logs say it's committing the terraform-docs change, but not able to push to your repo. If you aren't comfortable with that kind of a change, let me know.

**Comment by chrisgilmerproj** [comment]
I'm ok with setting up the ability for the tool to push back. I am following this doc here:

https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

It says I ought to see a way to allow editing below the `Notifications` on the right side but here's what I'm seeing:

Screen Shot 2021-08-03 at 2 44 50 PM

I'm wondering if there is an organizational block I've not figured out yet.

**Comment by dynamike** [comment]
That's likely it. This is what my debugging user interface looks like. I'm not sure what else to try at this point except maybe having the action just run on merges to `main`. It's an interesting edge case though. For now I'll merge and resolve the readme changes --

image

**Comment by chrisgilmerproj** [comment]
What an interesting find. Well, I guess I'm glad we at least tried. Thanks for helping me with this.

**Pull request merged**
